server/memstore.go

```diff
@@ -628,6 +628,61 @@ func (ms *memStore) LoadLastMsg(subject string) (subj string, seq uint64, hdr, m
 	return sm.subj, sm.seq, sm.hdr, sm.msg, sm.ts, nil
 }
 
+// LoadNextMsg will find the next message matching the filter subject starting at the start sequence.
+// The filter subject can be a wildcard.
+func (ms *memStore) LoadNextMsg(filter string, wc bool, start uint64) (subj string, seq uint64, hdr, msg []byte, ts int64, err error) {
+	ms.mu.RLock()
+	defer ms.mu.RUnlock()
+
```

Comment:
I see that in case of filestore, you check for "closed" state. Do you have the same for memstore?

Reply:
We do not check for closed state as much in memstore, just storing a new message.

Reply:
Will that cause unexpected and untested differences in behaviour?

Reply:
I don't think so since the upper layers are controlling access to the stores.